internal/services/users.go

```diff
@@ -96,3 +96,36 @@ func UserLogin(c *gin.Context) {
 	c.SetCookie("usrCookie", token, 86400, "/", "", true, true)
 	utils.SuccessWithData(c, user)
 }
+
+func CheckUser(c *gin.Context) {
```

Comment:
@paaart this is already handled in middleware. I feel this is redundant. Check and confirm